src/WorkItems/GitHubClientWrapper.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Octokit;
+using Octokit.Internal;
+
+namespace Microsoft.WorkItems
+{
+    /// <summary>
+    /// Represents a GitHub project in which work items can be filed.
+    /// </summary>
+    public class GitHubClientWrapper : FilingClient
+    {
+        private GitHubClient gitHubClient;
+
+        public override async Task Connect(string personalAccessToken)
+        {
+            var credentials = new Credentials(this.AccountOrOrganization, personalAccessToken);
+            var credentialsStore = new InMemoryCredentialStore(credentials);
+
+            await Task.Run(() =>
+            {
+                this.gitHubClient = new GitHubClient(new ProductHeaderValue(this.AccountOrOrganization), credentialsStore);
+            });
+        }
+
+        public override async Task<IEnumerable<WorkItemModel>> FileWorkItems(IEnumerable<WorkItemModel> workItemFilingMetadata)
+        {
+            foreach (WorkItemModel workItemModel in workItemFilingMetadata)
+            {
+                var newIssue = new NewIssue(workItemModel.Title)
+                {
+                    Body = workItemModel.BodyOrDescription,
+                };
+
+                Issue issue = await this.gitHubClient.Issue.Create(
+                    this.AccountOrOrganization, 
+                    this.ProjectOrRepository, 
+                    newIssue);
+
+                if (workItemModel.LabelsOrTags?.Count != 0)
+                {
+                    var issueUpdate = new IssueUpdate();
+
+                    foreach (string tag in workItemModel.LabelsOrTags)
+                    {
+                        issueUpdate.AddLabel(tag);
+                    }
+
+                    await this.gitHubClient.Issue.Update(
```

Comment:
>Update [](start = 50, length = 6)

Is there any API that would let you create the issue with all its data in a single operation? Or any "transactioning" feature that would let you wrap multiple operations? That would avoid situations where an exception causes the work item to end up in an incomplete state.

Reply:
Good question, there is no API of this kind that I can find. You can certainly create a bug with a label in the browser in a single operation (and fiddler suggests this is done in a single request). So maybe I missed something in the C# API. Will file a bug.

---
In reply to: [381455787](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381455787) [](ancestors = 381455787)